### ISSUE TITLE
Switching SAS mode while navball is hidden can crash programs

### ISSUE THREAD

**Issue by plonk75** [KSP-KOS/KOS#2831]
kOS V1.2.1.0
KSP 1.11.0.3045 (w/ Making History & Breaking Ground)

Steps to reproduce:
1. In VAB or SPH build a 3 part craft consisting of an OKTO2 core, a Z-200 battery and a CompoMax Radial Tubeless kOS processor. Align it vertically, so it won't start rolling when launched (SPH).
2. Launch it.
3. Open kOS terminal.
4. Switch SAS to on, either via console or via navball.
5. Hide navball via GUI.
6. Enter and execute` set sasmode to "normal"`. 

Result: 
Error message: `Object reference not set to an instance of an object`
Expected result:
No message, activation of Normal Mode for SAS.

7. Unhide navball.
8.  Enter and execute` set sasmode to "normal"`. 

Result: Expected result, no message, activation of Normal Mode for SAS.

Remarks:
Before testing, make sure the probe core supports the SAS mode to switch to, in the situation you test it in. 
The problem is not specific to the the mentioned above 3 parts, it occurs with other SAS modes, probe cores and kOS processors, too.

This one was pretty surprising when it ruined a video recording of an automated space plane flight, where I wanted to reduce screen clutter by hiding UI elements. After about an hour of recording (autom. launch, orbit, deorbit, land), it made the "bleepblop" sound, and the plane went for an uncontrolled lithobraking. :grin:

**Comment by JonnyOThan** [comment]
Your log file or at least the full exception stack would be useful.

I suspect this is actually related to functional changes in KSP 1.10 where they stopped updating or even removed certain UI elements when they are not visible.  I had similar issues with RPM in IVA mode because the navball stops updating.  If KOS depends on the SASMode buttons in the UI, I could easily see that breaking in later KSP versions and a simple recompile isn't going to help.

**Comment by nuggreat** [comment]
Here is my full log from KSP 1.10.1

[KSP.log](https://github.com/KSP-KOS/KOS/files/5771640/KSP.log)

Though the relevant exception looks to be 
```
[LOG 09:22:16.678] System.NullReferenceException: Object reference not set to an instance of an object
  at kOS.Module.kOSProcessor.SetAutopilotMode (System.Int32 mode) [0x00005] in <64b1c76e033842d2bd7fb92c02239116>:0 
  at kOS.Binding.FlightControlManager.SelectAutopilotMode (VesselAutopilot+AutopilotMode autopilotMode) [0x00059] in <64b1c76e033842d2bd7fb92c02239116>:0 
  at kOS.Binding.FlightControlManager.SelectAutopilotMode (System.String autopilotMode) [0x00196] in <64b1c76e033842d2bd7fb92c02239116>:0 
  at kOS.Binding.FlightControlManager.SelectAutopilotMode (System.Object autopilotMode) [0x00017] in <64b1c76e033842d2bd7fb92c02239116>:0 
  at kOS.Safe.Binding.BoundVariable.set_Value (System.Object value) [0x00015] in <1f05d5b75d1c42729d0e0281470dcabd>:0 
  at kOS.Safe.Execution.CPU.SetValue (System.String identifier, System.Object value) [0x00007] in <1f05d5b75d1c42729d0e0281470dcabd>:0 
  at kOS.Safe.Compilation.OpcodeStore.Execute (kOS.Safe.Execution.ICpu cpu) [0x00010] in <1f05d5b75d1c42729d0e0281470dcabd>:0 
  at kOS.Safe.Execution.CPU.ExecuteInstruction (kOS.Safe.Execution.ProgramContext context, System.Boolean doProfiling) [0x001e5] in <1f05d5b75d1c42729d0e0281470dcabd>:0 
  at kOS.Safe.Execution.CPU.ContinueExecution (System.Boolean doProfiling) [0x00030] in <1f05d5b75d1c42729d0e0281470dcabd>:0 
  at kOS.Safe.Execution.CPU.KOSFixedUpdate (System.Double deltaTime) [0x000b0] in <1f05d5b75d1c42729d0e0281470dcabd>:0 
```

The other interesting thing of note is that the SAS mode does change despite the crash.

**Comment by marianoapp** [comment]
I just checked the code and the SetAutopilotMode function is doing two things, changing the actual SAS mode and also toggling the UI button. This second step is the one failing.

**Comment by plonk75** [comment]
Damn, wrong click.

**Comment by 1greywind** [comment]
Yep, the issue persist in v1.3.2.0

**Comment by 1greywind** [comment]
> 

So adding try/catch around this line may help?

**Comment by Dunbaratu** [comment]
Closed by PR #3045, but I forgot to include the magic words that cause github to auto-associate the PR with the issue.  Manually closing this.
